### PR TITLE
Fix Docker builds for nerves_system_br v1.28.0 and later

### LINKED ISSUE #997: Docker permissions issue
### Environment

* Elixir version (`elixir -v`):
  ```
  Erlang/OTP 26 [erts-14.2.5] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [dtrace]

  Elixir 1.17.2 (compiled with Erlang/OTP 26)
  ```
* Nerves environment (`mix nerves.env --info`):
  ```
  ==> nerves
  ==> xxx_rpi5
  Generated xxx_rpi5 app
  |Elixir.Docker| Starting Build... (this may take a while)

  chown: changing ownership of '/nerves/build': Operation not permitted
  could not compile dependency :xxx_rpi5, "mix compile" failed.
  Errors may have been logged above.
  You can recompile this dependency with "mix deps.compile xxx_rpi5 --force",
  update it with "mix deps.update xxx_rpi5" or clean it with "mix deps.clean xxx_rpi5"
  ==> pnp_connect
  ** (Mix) The Nerves Docker build_runner encountered an error while setting permissions:

  %IO.Stream{device: :standard_io, raw: true, line_or_bytes: :line}
  ```
* Additional information about your host, target hardware or environment that may help:

  I suspect it might be connected to the move away from root, i.e. https://github.com/nerves-project/nerves_system_br/blob/v1.28.1/support/docker/nerves_system_br/Dockerfile#L137 vs. https://github.com/nerves-project/nerves/blob/main/lib/nerves/artifact/build_runners/docker.ex#L279. My understanding is that `chown` needs its `sudo` now.

### Current behaviour

```
% mix deps.compile xxx_rpi5 --force
==> nerves
==> xxx_rpi5
Generated xxx_rpi5 app
|Elixir.Docker| Starting Build... (this may take a while)

chown: changing ownership of '/nerves/build': Operation not permitted
could not compile dependency :xxx_rpi5, "mix compile" failed.
Errors may have been logged above.
You can recompile this dependency with "mix deps.compile xxx_rpi5 --force",
update it with "mix deps.update xxx_rpi5" or clean it with "mix deps.clean xxx_rpi5"
==> pnp_connect
** (Mix) The Nerves Docker build_runner encountered an error while setting permissions:

%IO.Stream{device: :standard_io, raw: true, line_or_bytes: :line}
```

### Expected behaviour

It compiles.

### PR DESCRIPTION
**Pull request by fhunleth**
Fixes #997

### CODE REVIEW

File: lib/nerves/artifact/build_runners/docker.ex [hunk 1]
```diff
@@ -64,7 +64,20 @@ defmodule Nerves.Artifact.BuildRunners.Docker do
 
   @version "~> 1.12 or ~> 1.12.0-rc2 or >= 17.0.0"
 
-  @working_dir "/nerves/build"
+  # nerves_system_br v1.8.0 changed to a Docker image that moves the
+  # project build directory and runs everything as a non-root user.
+  defp new_docker_image?() do
+    version_c = Application.spec(:nerves_system_br)[:vsn]
+    version_c != nil and Version.match?(to_string(version_c), "~> 1.28")
```
Comment (jjcarstens) [inline lib/nerves/artifact/build_runners/docker.ex:71]:
```suggestion
    version_c != nil and Version.match?(to_string(version_c), ">= 1.28")
```
I think this should cover all 1.x versions but do we can to make it any version higher than 1.28?

Comment (fhunleth) [reply lib/nerves/artifact/build_runners/docker.ex:71]: You mean cover 2.0 and higher? I'm totally for your change, but I think I'm misunderstanding the reason.

Comment (jjcarstens) [reply lib/nerves/artifact/build_runners/docker.ex:71]: Ya, that's what I mean. All versions 1.28 and higher including 2.0

**Comment by fhunleth** [comment]
@jjcarstens I think this PR works for nerves_system_br 1.28.x. At least it works on my one test. I'm getting this error with earlier systems and I think it's something else:

```
➜  nerves_system_bbb git:(v2.22.1) ✗ mix compile
==> nerves
Compiling 2 files (.ex)
Generated nerves app
==> nerves_system_bbb
Generated nerves_system_bbb app
|Elixir.Docker| Starting Build... (this may take a while)

** (Mix) The Nerves Docker build_runner encountered an error while building:

-----
Setting User
UID: 501
GID: 20
useradd: warning: the home directory /home/nerves already exists.
useradd: Not copying any file from skel directory into it.
Switching user
error: exec: "/nerves/env/platform/create-build.sh": stat /nerves/env/platform/create-build.sh: no such file or directory

-----
```

File: lib/nerves/artifact/build_runners/docker.ex [hunk 2]
```diff
@@ -64,7 +64,20 @@ defmodule Nerves.Artifact.BuildRunners.Docker do
 
   @version "~> 1.12 or ~> 1.12.0-rc2 or >= 17.0.0"
 
-  @working_dir "/nerves/build"
+  # nerves_system_br v1.8.0 changed to a Docker image that moves the
```
[the remark this answers is not in the document]
Comment (fhunleth) [reply lib/nerves/artifact/build_runners/docker.ex:67]: Yes, thanks for catching that.

**Comment by fhunleth** [comment]
Now I can't reproduce the issue I was seeing. I upgraded Docker on my laptop since I had an old version, and I cleaned the build directory. Maybe it was the latter. 

I'm going to mark this as ready for review since it seems to work.

**Review by jjcarstens — APPROVED** [review]

**Pull request merged**